src/Apps/Ingress/Pages/IngressApp/IngressApp.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useCallback, useReducer } from 'react';
+import {
+    IIngressAppProps,
+    IIngressAppStyleProps,
+    IIngressAppStyles
+} from './IngressApp.types';
+import { getStyles } from './IngressApp.styles';
+import { classNamesFunction, IconButton, styled } from '@fluentui/react';
+import { useExtendedTheme } from '../../../../Models/Hooks/useExtendedTheme';
+import BaseComponent from '../../../../Components/BaseComponent/BaseComponent';
+import HomePage from '../../Components/HomePage/HomePage';
+import {
+    NavigationContext,
+    NavigationContextReducer
+} from '../../Models/Context/NavigationContext';
+import { PageNames } from '../../Models/Constants';
+import { NavigationContextActionType } from '../../Models/Context/NavigationContext.types';
+
+const getClassNames = classNamesFunction<
+    IIngressAppStyleProps,
+    IIngressAppStyles
+>();
+
+const IngressApp: React.FC<IIngressAppProps> = (props) => {
+    const { adapter, locale, localeStrings, styles, theme } = props;
+
+    // contexts
+
+    // state
+    const [navigationContextState, navigationContextDispatch] = useReducer(
+        NavigationContextReducer,
+        {
```

Comment:
suggestion: You might want to define the default state in some file so you can use it in stories and in the real app. This state might end up getting kinda long.